crates/artifacts/resolc/src/contract.rs

```diff
@@ -47,17 +47,31 @@ pub struct ResolcContract {
 impl From<ResolcContract> for foundry_compilers_artifacts_solc::Contract {
     fn from(contract: ResolcContract) -> Self {
         let meta = match contract.metadata.as_ref() {
-            Some(meta) => match meta {
-                serde_json::Value::Object(map) => {
-                    if let Some(meta) = map.get("solc_metadata") {
-                        serde_json::from_value::<LosslessMetadata>(meta.clone()).ok()
-                    } else {
-                        None
-                    }
-                }
-                _ => None,
-            },
-            None => Default::default(),
+            Some(serde_json::Value::Object(map)) => {
+                map.get("solc_metadata")
+                    .and_then(|solc_metadata| {
+                        serde_json::from_value::<LosslessMetadata>(solc_metadata.clone()).ok()
+                    })
+                    .map(|mut solc_metadata| {
+                        // Extract and inject revive compiler information if available.
+                        if let (Some(revive_version), Some(solc_version)) = (
+                            map.get("revive_version").and_then(|v| v.as_str()),
+                            map.get("solc_version").and_then(|v| v.as_str()),
+                        ) {
+                            // Update version and regenerate raw metadata.
+                            solc_metadata.metadata.compiler.version = format!(
```

Comment:
why? `Compiler.version` is a string, not an object. if at some point `raw_metadata` to be used it can break things downstream, have you checked that?.(i haven't actually looked whether it's used yet.)